pixeltable/tests/test_dataframe.py

```diff
@@ -110,3 +115,163 @@ def test_select_literal(self, test_tbl: catalog.Table) -> None:
         t = test_tbl
         res = t.select(1.0).where(t.c2 < 10).show(0)
         assert res.rows == [[1.0]] * 10
+
+    def test_to_pytorch_dataset(self, all_datatype_tbl: catalog.Table):
+        """ tests all types are handled correctly in this conversion
+        """
+        import torch
+
+        t = all_datatype_tbl
+        df = t.where(t.row_id < 1)
+        assert df.count() > 0
+        ds = df.to_pytorch_dataset()
+        type_dict = dict(zip(df.get_column_names(),df.get_column_types()))
+        for tup in ds:
+            for col in df.get_column_names():
+                assert col in tup
+        
+            arrval = tup['c_array']
+            assert isinstance(arrval, np.ndarray)
+            col_type = type_dict['c_array']
+            assert arrval.dtype == col_type.numpy_dtype()
+            assert arrval.shape == col_type.shape
+            assert arrval.dtype == np.float32
+
+            assert isinstance(tup['c_bool'], bool)
+            assert isinstance(tup['c_int'], int)
+            assert isinstance(tup['c_float'], float)
+            assert isinstance(tup['c_timestamp'], float)
+            assert torch.is_tensor(tup['c_image'])
+            assert isinstance(tup['c_video'], str)
+            assert isinstance(tup['c_json'], dict)
+
+    def test_to_pytorch_image_format(self, all_datatype_tbl: catalog.Table) -> None:
+        """ tests the image_format parameter is honored
+        """
+        import torch
+
+        t = all_datatype_tbl
+        df = t.select(
+            t.row_id,
+            t.c_image,
+            c_image_xformed=t.c_image.resize([220, 224]).convert('RGB')
+        ).where(t.row_id < 1)
+
+        ds = df.to_pytorch_dataset(image_format='np')
+        ds_ptformat = df.to_pytorch_dataset(image_format='pt')
+
+        elts = []
+        for elt, elt_pt in zip(ds, ds_ptformat):
+            arr = elt['c_image']
+            assert isinstance(arr, np.ndarray)
+
+            arr = elt['c_image_xformed']
+            assert isinstance(arr, np.ndarray)
+            assert arr.shape == (224, 220, 3)
+            assert arr.dtype == np.uint8
+
+            arr_pt = elt_pt['c_image']
+            assert torch.is_tensor(arr_pt)
+
+            arr_pt = elt_pt['c_image_xformed']
+            assert torch.is_tensor(arr_pt)
+            assert arr_pt.shape == (3, 224, 220)
+            assert arr_pt.dtype == torch.float32
+            assert (0.0 <= arr_pt).all()
+            assert (arr_pt <= 1.0).all()
```

Comment:
Is there some way to check that the c_image and c_image_xformed values are "the same" (within some epsilon)? If it's difficult, leave to-do. If it's easy, add it.